kotlinpoet-classinspector-reflective/src/main/kotlin/com/squareup/kotlinpoet/classinspector/reflective/ReflectiveClassInspector.kt

```diff
@@ -46,11 +46,14 @@ import java.lang.reflect.Field
 import java.lang.reflect.Method
 import java.lang.reflect.Modifier
 import java.lang.reflect.Parameter
+import java.net.URLClassLoader
 import java.util.concurrent.ConcurrentHashMap
 import kotlin.LazyThreadSafetyMode.NONE
 
 @KotlinPoetMetadataPreview
-public class ReflectiveClassInspector private constructor() : ClassInspector {
+public class ReflectiveClassInspector private constructor(
+  private val classLoader: URLClassLoader?
```

Comment:
Any reason to type this as `URLClassLoader` and not `ClassLoader`? Will a different implementation of `ClassLoader` work here?

Reply:
@Egorand I changed this to ClassLoader in the following commit